apache2/apache2_util.c

```diff
@@ -336,11 +334,17 @@ static void send_waf_log(struct waf_lock* lock, apr_file_t** fd, const char* str
     get_field_value("[line ", "]", str1, waf_line);
     get_field_value("[msg ", "]", str1, waf_message);
     get_field_value("[data ", "]", str1, waf_data);
-	get_field_value("[ver ", "]", str1, waf_ruleset_info);
 	get_ip_port(ip_port, waf_ip, waf_port);
     get_detail_message(str1, waf_detail_message); 
     get_short_filename(waf_filename);
-    get_ruleset_type_version(waf_ruleset_info, waf_ruleset_type, waf_ruleset_version); 
+    get_ruleset_type_version(waf_signature, waf_ruleset_type, waf_ruleset_version); 
+
+    // overwrite message with pcre limits exceeded message if the the orig_string contained a pcre limits exceeded message
+    if (str1 && (strstr(str1, " Execution error - PCRE limits exceeded ") != NULL)) {
```

Comment:
this is a costly operation for datapath, if we know the format of the error msg we can maybe limit the search to the first X chars

Reply:
@alonzop  has a valid point here, since we are doing an O(N) search on every log message that is getting generated.  If there any way of passing an error code here that pre-specifies that the error is because of PCRE Limits being exceeded?